lua/telescope/_extensions/file_browser/make_entry.lua

```diff
@@ -191,6 +204,22 @@ local make_entry = function(opts)
       return raw
     end
 
+    if k == "file_status" then
+      local file_statuses = vim.F.if_nil(opts.file_statuses, {})
```

Comment:
Here, we'd create an empty table whenever there is no git status but it's accessed for a file (not sure when that happens, but a bit of an anti pattern).

Something along these lines should probably be moved to the top of the function (before the closure where the entry maker is returned)

```lua
opts.git_file_status = vim.F.if_nil(opts.git_file_status, {})
```

Reply:
> not sure when that happens, but a bit of an anti pattern

Initially I was thinking that the async case wouldn't pass in any git statuses, but I changed that to now pass in an empty table there, so this is theoretically not required as it should always be there. I can still make it nil-safe with your suggestion though. What do you prefer? Delete or add your suggestion to the top of the function?